lib/layers/data/repositories/repository_impl.dart

```diff
@@ -0,0 +1,242 @@
+    return Future.wait([
+      localSource.fetchAchievementsById(achievementId, EPIC),
+      localSource.fetchAchievementsById(achievementId, ACTION),
+      localSource.fetchAchievementsById(achievementId, SPECIAL),
+      localSource.fetchAchievementsById(achievementId, MEMORIAL),
+      localSource.fetchAchievementsById(achievementId, GROUP),
+      localSource.fetchAchievementsById(achievementId, CLASS),
+    ]);
+  }
+
+  @override
+  Future<PersonalData> fetchPersonalData() async {
+    final PersonalDataApi personalDataApi;
+    final ClanInfo? clanInfo;
+    try {
+      personalDataApi = await _fetchPersonalDataApi(signedUser);
+      final int? clanId =
+          personalDataApi.data?[signedUser.id.toString()]?.clanId;
+      clanInfo = (clanId != null)
+          ? await remoteSource.fetchClanInfo(clanId: clanId)
+          : null;
+    } catch (e) {
+      throw Exception('Check internet connection');
+    }
+    return PersonalData.fromPersonalAndClanInfo(personalDataApi, clanInfo);
+  }
+
+  Future<PersonalDataApi> _fetchPersonalDataApi(UserData signedUser) async {
+    final PersonalDataApi personalDataApi;
+    try {
+      personalDataApi = await remoteSource.fetchPersonalData(
+        accountId: signedUser.id,
+        accessToken: signedUser.accessToken,
+      );
+    } catch (e) {
+      throw Exception('Check internet connection');
+    }
+    return personalDataApi;
+  }
+
+  @override
+  Future<List<Vehicle>> fetchUserVehicles() async {
+    final UserVehiclesApi userVehicles = await remoteSource.fetchUserVehicles(
+      accountId: signedUser.id,
+      accessToken: signedUser.accessToken,
+    );
+    final List<int> tankIds = userVehicles.createListOfTankId();
+    final List<VehiclesDataTTC> vehiclesByIdFromDb =
+        await localSource.fetchTTCByListOfIDs(tankIds);
+    final List<Vehicle> result = _createVehicleListFrom(
+        vehiclesByIdFromDb, userVehicles.vehicles.values.first);
+    return result;
+  }
+
+  List<Vehicle> _createVehicleListFrom(List<VehiclesDataTTC> vehiclesByIdFromDb,
+      List<UserVehicle> userVehicles) {
+    List<Vehicle> result = [];
+    for (var i = 0; i < vehiclesByIdFromDb.length; i++) {
+      result
+          .add(Vehicle.fromTtcAndUser(userVehicles[i], vehiclesByIdFromDb[i]));
+    }
+    return result;
+  }
+
+  Future<void> initOrSyncVehiclesDatabase() async {
+    //todo language!!!!!!!!!
+    baseOptions.baseUrl = BASE_URL_EU; //todo from shared prefs
+    Future.wait([
+      _initVehiclesDatabase(),
+      _initAchievesDatabase(),
+    ]);
+  }
+
+  Future<void> _initAchievesDatabase() async {
+    final int achievesInDbCount = localSource.getAchievesCount();
+    final AchievementsDataBase achievementsDataBase;
+    try {
+      achievementsDataBase = await remoteSource.fetchAchievesDataBase();
+    } catch (e) {
+      throw Exception('Check internet connection');
+    }
+    if (achievesInDbCount == achievementsDataBase.meta.count) return;
+    // else create DB
```

Comment:
Comments is a bad practice, to simplify the code you can use extracting into function, eg `createDatabase()`